plugins/secretstores/docker/docker_test.go

```diff
@@ -0,0 +1,32 @@
+package docker
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSampleConfig(t *testing.T) {
+	plugin := &Docker{}
+	require.NotEmpty(t, plugin.SampleConfig())
+}
+
+func TestInitFail(t *testing.T) {
+	tests := []struct {
+		name     string
+		plugin   *Docker
+		expected string
+	}{
+		{
+			name:     "invalid id",
+			plugin:   &Docker{},
+			expected: "id missing",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.plugin.Init()
+			require.ErrorContains(t, err, tt.expected)
+		})
+	}
+}
```

Comment:
Please add a test-case with some dummy files. To do this it might be necessary to make the `dockerSecretsDir` a variable, a plugin variable or to be able to prefix it somehow e.g. using environment variables. This way, you can add some dummy files in a `testdata` subdir and run tests against those files / file-sets.

Reply:
I would add a `path` parameter to the `sample.conf` which will have the value `/run/secrets` and a Note that this is currently the default path for docker. If there are cases, where users specifically mount files to different directories, the `path` can be set to desired directory and the plugin should work fine too.

In this case, most of the `jose` test cases will be used with the exception of `Set` since under `/run/secrets` is a read-only directory.